src/librawspeed/decoders/DngDecoder.cpp

```diff
@@ -767,9 +767,12 @@ bool DngDecoder::decodeBlackLevels(const TiffIFD* raw) const {
   iPoint2D blackdim(1, 1);
   if (raw->hasEntry(TiffTag::BLACKLEVELREPEATDIM)) {
     const TiffEntry* bleveldim = raw->getEntry(TiffTag::BLACKLEVELREPEATDIM);
-    if (bleveldim->count != 2)
+    if (bleveldim->count == 1)
+      blackdim = iPoint2D(bleveldim->getU32(0), bleveldim->getU32(0));
+    else if (bleveldim->count == 2)
+      blackdim = iPoint2D(bleveldim->getU32(0), bleveldim->getU32(1));
+    else
       return false;
```

Comment:
```
if (bleveldim->count == 2)
  blackdim = iPoint2D(bleveldim->getU32(0), bleveldim->getU32(1));
else if (bleveldim->count == 1) {
  blackdim = iPoint2D(bleveldim->getU32(0), bleveldim->getU32(0));
  if (bleveldim != {1,1})
    return false;
  writeLog("???");
else
  return false;
}
```

Reply:
> Why do you assume a single value must denote (1,1)?

Why do you assume a single value must denote (N,N)?
Maybe it means (N/2,N/2) or something?

> Why do you assume a single value must denote (1,1)?

I'm not assuming that. I'm just saying that this workaround
is just completely horrible, and i'm not okay with
blindly accepting random crap values, only `(1)`,
because it's rather unambigious.

> What's the difference then to the previous version falling back to default (1,1)?

There we'd also accept obviously-broken 3-component size.

Reply:
Ok, let's cover this particular case then only...